cscs-checks/apps/espresso/espresso_check.py

```diff
@@ -15,9 +15,15 @@ def __init__(self, variant, **kwargs):
                                        'Espresso')
         self.executable = 'pw.x'
         self.executable_opts = '-in ausurf.in'.split()
-
-        self.sanity_patterns = sn.assert_found(
-            r'convergence has been achieved', self.stdout)
+        property_patterns = {
+            'energy': sn.extractsingle(r'!    total energy              ='
+                                       r'\s+(?P<energy>\S+) Ry',
+                                       self.stdout, 'energy', float)
+        }
+        self.sanity_patterns = sn.all([
+            sn.assert_found(r'convergence has been achieved', self.stdout),
+            sn.assert_reference(property_patterns['energy'], -11427.08612278,
+                                lower_thres=-1e-10, upper_thres=1e-10)])
```

Comment:
No need to use the `lower_thres` and `upper_thres` names, since lower thresholds are always negative and upper thresholds are always positive in `assert_reference()`.